pkg/tsdb/testdata/scenarios.go

```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
 
 	queryRes := tsdb.NewQueryResult()
 	queryRes.Series = append(queryRes.Series, series)
+	attachLabels(query, queryRes)
 	return queryRes
 }
 
+func attachLabels(query *tsdb.Query, queryRes *tsdb.QueryResult) {
+	labelText := query.Model.Get("labels").MustString("")
+	if labelText == "" {
+		return
+	}
+
+	re := regexp.MustCompile(`\b(\w+)(!?=~?)"([^"\n]*?)"`)
+	tags := make(map[string]string)
+
+	byt := []byte(labelText)
+	re.ReplaceAllFunc(byt, func(s []byte) []byte {
+		exp := string(s)
+		idx := strings.Index(exp, "=")
+		key := exp[:idx]
+		val := util.TrimQuotes(exp[idx+1:])
+		tags[key] = val
+		return s
+	})
+	for _, series := range queryRes.Series {
+		series.Tags = tags
```

Comment:
I wonder what would break if we change this to Labels :)   (in json response).

Reply:
it is also in gRPC, so I don't really want to think about it ;)

The backend signatures should stay the same for TableData and SeriesData...  when we add DataFrame, we can start to migrate them all to a consistent layout